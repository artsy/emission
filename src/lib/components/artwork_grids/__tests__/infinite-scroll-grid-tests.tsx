import 'react-native'

import * as React from 'react'
import { renderWithLayout } from '../../../tests/render_with_layout'

import InfiniteScrollArtworksGrid from '../infinite_scroll_grid'

jest.mock('../../../metaphysics.ts')
jest.mock('../../opaque_image_view.tsx', () => 'AROpaqueImageView')
jest.mock('../../spinner.tsx', () => 'ARSpinner')

it('renders properly', () => {
  const artworks = [ artwork(), artwork(), artwork() ]

  const grid = renderWithLayout(<InfiniteScrollArtworksGrid artworks={artworks} />, { width: 768})
  expect(grid).toMatchSnapshot()
})

var artwork = () => {
  return {
    __id: Math.random(),
    title: 'TALKING POLITICS',
    date: '2008',
    sale_message: 'Not for sale',
    is_in_auction: false,
    image: {
      url: 'artsy.net/image-url',
      aspect_ratio: 1.27
    },
    artists: [
      {
        name: 'Jenny Holzer'
      }
    ],
    partner: {
      name: 'Kukje Gallery'
    },
    href: '/artwork/jenny-holzer-talking-politics'
  }
}
