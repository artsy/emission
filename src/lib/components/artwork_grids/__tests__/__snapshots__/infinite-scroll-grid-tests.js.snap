exports[`test renders properly 1`] = `
<ScrollView
  accessibilityLabel="Artworks ScrollView"
  onLayout={[Function]}
  onScroll={[Function]}
  scrollEventThrottle={50}
  scrollsToTop={false}>
  <View
    accessibilityLabel="Artworks Content View"
    style={
      Object {
        "flexDirection": "row",
      }
    }>
    <View
      accessibilityLabel="Section 0"
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "marginRight": 20,
            "width": 374,
          },
        ]
      }>
      <TouchableWithoutFeedback
        onPress={[Function]}>
        <View>
          <AROpaqueImageView
            aspectRatio={1.27}
            imageURL="artsy.net/image-url"
            style={
              Object {
                "marginBottom": 10,
              }
            } />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Array [
                  Object {
                    "color": "#666666",
                    "fontSize": 12,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ],
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Jenny Holzer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Array [
                    Object {
                      "color": "#666666",
                      "fontSize": 12,
                    },
                    Object {
                      "fontStyle": "italic",
                    },
                  ],
                  Object {
                    "fontFamily": "AGaramondPro-Regular",
                  },
                ]
              }>
              TALKING POLITICS
            </Text>
            , 2008
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Kukje Gallery
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Not for sale
          </Text>
        </View>
      </TouchableWithoutFeedback>
      <View
        accessibilityLabel="Spacer View"
        style={
          Object {
            "height": 20,
          }
        } />
      <TouchableWithoutFeedback
        onPress={[Function]}>
        <View>
          <AROpaqueImageView
            aspectRatio={1.27}
            imageURL="artsy.net/image-url"
            style={
              Object {
                "marginBottom": 10,
              }
            } />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Array [
                  Object {
                    "color": "#666666",
                    "fontSize": 12,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ],
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Jenny Holzer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Array [
                    Object {
                      "color": "#666666",
                      "fontSize": 12,
                    },
                    Object {
                      "fontStyle": "italic",
                    },
                  ],
                  Object {
                    "fontFamily": "AGaramondPro-Regular",
                  },
                ]
              }>
              TALKING POLITICS
            </Text>
            , 2008
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Kukje Gallery
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Not for sale
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <View
      accessibilityLabel="Section 1"
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "marginRight": 0,
            "width": 374,
          },
        ]
      }>
      <TouchableWithoutFeedback
        onPress={[Function]}>
        <View>
          <AROpaqueImageView
            aspectRatio={1.27}
            imageURL="artsy.net/image-url"
            style={
              Object {
                "marginBottom": 10,
              }
            } />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Array [
                  Object {
                    "color": "#666666",
                    "fontSize": 12,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ],
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Jenny Holzer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Array [
                    Object {
                      "color": "#666666",
                      "fontSize": 12,
                    },
                    Object {
                      "fontStyle": "italic",
                    },
                  ],
                  Object {
                    "fontFamily": "AGaramondPro-Regular",
                  },
                ]
              }>
              TALKING POLITICS
            </Text>
            , 2008
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Kukje Gallery
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#666666",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }>
            Not for sale
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  </View>
</ScrollView>
`;
